openwisp_users/base/models.py

```diff
@@ -109,6 +109,23 @@ def is_owner_of_any_organization(self):
                 return True
         return False
 
+    def selected_org(self, org_field):
+        if self.is_superuser:
+            cache_key = 'selected_{}_org'.format('superuser')
+        else:
+            cache_key = 'selected_{}_org'.format(self.pk)
+        selected_org = cache.get(cache_key)
+        if selected_org is not None:  # If cache is not deleted
+            if selected_org is False:
```

Comment:
when there's no org to select, set `selected_org` to None so that this second if while not be needed

Reply:
If you will notice carefully then you will realise that if we set it to None then query will be evaluated for None. I want this function to return None or an org. That's why I am storing False in place of None which will indicate it have to return None and query will not be evaluated for None always.

Reply:
@nemesisdesign is this cache at the right place? It makes more sense to me if it was on the `organization model`

Reply:
I am making cache key differently for superuser because of following reason. 

1. When an org is deleted then Org Owner and Org User is also deleted which sends a signal that is received in `apps.py` file. There in instance we get user who is associated with the deleted org but superuser is not associated with any org. So we can't delete cache stored for superuser. 

I preferred to add cache here because we need to have cache for every user.